src/cpu_feature/arm_linux/cpuinfo.rs

```diff
@@ -0,0 +1,153 @@
+use std::collections::HashMap;
+#[cfg(all(any(target_arch = "arm", target_arch = "aarch64"), target_os="linux"))]
+use std::fs::File;
+#[cfg(all(any(target_arch = "arm", target_arch = "aarch64"), target_os="linux"))]
+use std::io::BufReader;
+use std::io::BufRead;
+use std::string::{String, ToString};
+
+#[derive(Debug, PartialEq)]
+pub enum CpuInfoError {
+    IoError,
+    InvalidFormat
+}
+
+pub type CpuInfo = HashMap<String, String>;
+
+#[cfg(all(any(target_arch = "arm", target_arch = "aarch64"), target_os="linux"))]
+pub fn parse_cpuinfo() -> Result<CpuInfo, CpuInfoError> {
+    match File::open("/proc/cpuinfo").map(|f| BufReader::new(f)) {
+        Ok(mut r) => parse_cpuinfo_reader(&mut r),
+        Err(_) => Err(CpuInfoError::IoError)
+    }
+}
+
+/// parse the contents of /proc/cpuinfo into a map of field names to values.
+/// Keeps the first encountered value for each name.
+#[cfg(any(test, target_os="linux"))]
+pub fn parse_cpuinfo_reader(input: &mut BufRead)
```

Comment:
I could change it to take a Path instead of a BufRead and convert the tests that use string literals to all have their own files instead, but I'll still need *something* that's not just "read from exactly /proc/cpuinfo" to be able to test it.

The reason it's public is that it's used in testing the higher-level armcap logic with various cpuinfo files. Is there a better way to expose that to a test submodule of another module?